tests/test_process.py

```diff
@@ -1106,6 +1106,142 @@ def test_process_signal_min_max(
     pd.testing.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    'process_func, signal, sampling_rate',
+    [
+        (
+            lambda x, sr: x.mean(),
+            np.array([0, 0, 0, 0, 1, 1, 1, 1], dtype=np.float32),
+            1,
+        ),
+    ]
+)
+@pytest.mark.parametrize(
+    'start, end, win_dur, hop_dur, expected',
+    [
+        (
+            None, None, 4, None,
+            pd.Series(
+                [0, 0.5, 1],
+                audinterface.utils.signal_index(
+                    [0, 2, 4],
+                    [4, 6, 8],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            None, None, 4, 2,
+            pd.Series(
+                [0, 0.5, 1],
+                audinterface.utils.signal_index(
+                    [0, 2, 4],
+                    [4, 6, 8],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, None, 4, 2,
+            pd.Series(
+                [0.25, 0.75],
+                audinterface.utils.signal_index(
+                    [1, 3],
+                    [5, 7],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, 5.0, 4, 2,
+            pd.Series(
+                [0.25],
+                audinterface.utils.signal_index(1, 5),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, 2.0, 4, 2,
```

Comment:
Could you also add a test for the case where the window is long enough, but `start` is out of bound, e.g. `9.0, 15.0, 4, 2`?

Reply:
Done.

Reply:
Thanks, I was also curious to see if it fails as `start` and `end` are out of bound or if it returns an empty array. Seems to be the later. Is this behavior the same, when not using a sliding window, or do we then get an error for out of bound `start` and `end` values?

Reply:
> when not using a sliding window, or do we then get an error for out of bound start and end values

In that case an empty array is passed to the processing function so it depends how it is handled by the function.

Reply:
Makes sense